src/app/auth/login/login.component.ts

```diff
@@ -27,36 +33,150 @@ export class LoginComponent {
+    );
+  }
+
+  public get registrationUsernameReq() {
+    return this.registrationForm
+      .get('usernameRegistration')
+      ?.hasError('required');
+  }
+
+  public get registrationUsernameMax() {
+    return this.registrationForm
+      .get('usernameRegistration')
+      ?.hasError('maxlength');
+  }
+
+  public get registrationUsernameMin() {
+    return this.registrationForm
+      .get('usernameRegistration')
+      ?.hasError('minlength');
+  }
+
+  public get registrationPasswordValid() {
+    return (
+      this.registrationForm.get('passwordRegistration')!.hasError('pattern') &&
+      !this.registrationForm.get('passwordRegistration')!.hasError('required')
+    );
+  }
+
+  public get registrationPasswordMin() {
+    return this.registrationForm
+      .get('passwordRegistration')!
+      .hasError('minlength');
+  }
+
+  public get registrationPasswordReq() {
+    return this.registrationForm
+      .get('passwordRegistration')!
+      .hasError('required');
+  }
+
   constructor(
     private fb: FormBuilder,
     private auth: AuthService,
     private router: Router
   ) {
     this.loginForm = this.fb.group({
-      username: new FormControl(''),
-      email: new FormControl(''),
-      password: new FormControl(''),
+      emailLogin: new FormControl('', [
+        Validators.required,
+        Validators.pattern('^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+$'),
+      ]),
+      passwordLogin: new FormControl('', [
+        Validators.minLength(5),
+        Validators.required,
+        Validators.pattern('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])[a-zA-Z0-9]+$'),
+      ]),
+    });
+
+    this.registrationForm = this.fb.group({
+      usernameRegistration: new FormControl('', [
+        Validators.maxLength(12),
+        Validators.minLength(5),
+        Validators.pattern('^[a-z0-9_-]+$'),
+        Validators.required,
+      ]),
+      emailRegistration: new FormControl('', [
+        Validators.required,
+        Validators.pattern('^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+$'),
+      ]),
+      passwordRegistration: new FormControl('', [
+        Validators.minLength(5),
+        Validators.required,
+        Validators.pattern('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])[a-zA-Z0-9]+$'),
+      ]),
     });
   }
 
   async registration(): Promise<void> {
     const user: IUser = {
       id: '',
-      username: this.loginForm.get('username')!.value,
-      email: this.loginForm.get('email')!.value,
+      username: this.registrationForm.get('usernameRegistration')!.value,
+      email: this.registrationForm.get('emailRegistration')!.value,
       profilePhoto: '',
       createdOn: new Date(),
     };
 
-    await this.auth.signUp(user, this.loginForm.get('password')!.value);
-    this.router.navigate(['/board']);
+    if (
+      (await this.auth.signUp(
+        user,
+        this.registrationForm.get('passwordRegistration')!.value
+      )) === true
+    ) {
```

Comment:
Please move a condition to a separate method and property or consider to create a login/registration guard